### PR TITLE
Make project naming consistent in documentation and respect licensing

### PR DESCRIPTION
**Pull request by zenseii**
This changes documentations so that naming of the project is consistent throughout and so that it also very importantly respects trademark and licenses.

### CODE REVIEW

**Comment by zenseii** [comment]
Hi @ihhub , @oleg-derevenetz  and @idshibanov ! Would you have the time to look at this?

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # fheroes2
 
-**Free Heroes of Might and Magic II** (fheroes2) is a recreation of HoMM2 game engine.
+**fheroes2 Resurrection** (fheroes2) is a recreation of HoMM2 game engine.
```
Comment (ihhub) [inline README.md:3]: `**fheroes2** is a recreation of Heroes of Might and Magic II game engine.`

File: docs/INSTALL.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 ## Requirements
 
-You will need to have a demo version or the full version of **Heroes of Might and Magic II** game to be able to play. We strongly
```
Comment (ihhub) [inline docs/INSTALL.md:5]: Please revert this change.

File: docs/README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # fheroes2
 
-**Free Heroes of Might and Magic II** (fheroes2) is a recreation of HoMM2 game engine.
```
Comment (ihhub) [inline docs/README.md:3]: `**fheroes2** is a recreation of Heroes of Might and Magic II game engine.`

File: docs/README.txt [hunk 4]
```diff
@@ -11,10 +11,10 @@ using one of the bundled scripts.
 You will need to have a demo version or the full version of Heroes of Might and
 Magic II game to be able to play.
 
-If you do not have the original Heroes of Might and Magic II game, run the
```
Comment (ihhub) [inline docs/README.txt:14]: Please revert changes in this file.

File: docs/README_PSV.md [hunk 5]
```diff
@@ -3,7 +3,7 @@
 ## Install
 Install fheroes2.vpk to your Vita.
 
-fheroes2 requires data files from the original Heroes of Might and Magic II.
```
Comment (ihhub) [inline docs/README_PSV.md:6]: Please revert changes in this file.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @zenseii , from licensing point of view only usage of Heroes of Might and Magic II phrase is not permitted in a name of the project while in description and documentation we have to use the official name of the original game.

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@zenseii , thank you so much for these changes!
